gossipd/gossip.c

```diff
@@ -1054,13 +1054,21 @@ static struct io_plan *getroute_req(struct io_conn *conn, struct daemon *daemon,
 	u16 riskfactor;
 	u8 *out;
 	struct route_hop *hops;
-
-	fromwire_gossip_getroute_request(msg, NULL, &source, &destination,
-					 &msatoshi, &riskfactor, &final_cltv);
+	double inefficiency;
+	u32 inefficiency_scaled;
+	u8 *seed;
+
+	fromwire_gossip_getroute_request(tmpctx, msg, NULL,
+					 &source, &destination,
+					 &msatoshi, &riskfactor, &final_cltv,
+					 &inefficiency_scaled, &seed);
+	inefficiency = ((double) inefficiency_scaled) / 4294967296.0;
```

Comment:
Like @rustyrussell I'd prefer having an integer percentage that doesn't require scaling up and down.